### PR TITLE
Enrichment integration

### PR DESCRIPTION
**Pull request by d2fong**
- integrate enrichment app into the search page
- use generic app logos for the apps
- fix xref link overflow refs #1165 
- fix long titles #1232 

screen shot 2018-12-11 at 4 50 04 pm

### CODE REVIEW

**Comment by d2fong** [comment]
We decided from some discussions earlier that showing an enrichment app with only < 3 genes doesn't seem that useful and showing the interactions app for > 3 genes doesn't seem that useful either.

**Comment by d2fong** [comment]
what do you mean by vertical and horizontal controls?

**Comment by d2fong** [comment]
I agree about the 'top 25' genes.  We don't actually do any work to determine the top 25 genes so its false advertising.

**Comment by maxkfranz** [comment]
I think the cards look really good.  I'll review the code asap

File: src/client/features/enrichment/index.js [hunk 1]
```diff
@@ -6,9 +6,7 @@ const classNames = require('classnames');
 const queryString = require('query-string');
 
 const EnrichmentToolbar = require('./enrichment-toolbar');
-const EmptyNetwork = require('../../common/components/empty-network');
-const PcLogoLink = require('../../common/components/pc-logo-link');
-const CytoscapeNetwork = require('../../common/components/cytoscape-network');
+const { EmptyNetwork, PcLogoLink, CytoscapeNetwork } = require('../../common/components/');
```
Comment (jvwong) [inline src/client/features/enrichment/index.js]: While we're here: I don't think this `EmptyNetwork` component is shown even if there is no content (i.e. nodes.length) unlike `features/interactions`.

**Comment by jvwong** [comment]
Just to reiterate our discussion there are two requirements: We need to convey what the app will actually provide (e.g. pathway enrichment) and the input we're are seeding that app with (i.e. HGNC gene IDs). 

For multi-gene case, I was thinking about a setup where the genes are a thin strip, scrollable horizontally, and then maybe we finally use the margin for the 'app' panel, where we can put more of a full text description:
- Interactions
  - Blurb: `Interactions displays a network visualization of Gene Ontology (Biological Process) and Reactome pathways that match some of the genes in your query.`
  - Button: `View genes that interact with X`
- Enrichment
  - Blurb:`Enrichment displays a network visualization of Gene Ontology (Biological Process) and Reactome pathways that match some of the genes in your query.`
  - Button: `View enriched pathways.`

Of course this is inspired by the Googles: 

image

File: src/client/features/search/gene-results-view.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  const hgncXrefLink = _.find( xrefLinks, link  => link.namespace === NS_HGNC_SYMBOL );
+  if( hgncXrefLink ) symbol = _.last( _.compact( hgncXrefLink.uri.split('/') ) );
+  return symbol;
+};
+
+class EntitySummaryBox extends React.Component {
+  render(){
+    let { summary } = this.props;
+    let { displayName, xrefLinks } = summary;
+    const hgncSymbol = getHgncFromXref( xrefLinks );
```
Comment (jvwong) [inline src/client/features/search/gene-results-view.js]: `props` will now contain values for "geneSymbol" so we can ditch `getHgncFromXref`

File: src/client/features/search/gene-results-view.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    let { geneResults } = this.props;
+    let sources = geneResults.map( summaryResult => getHgncFromXref( _.get( summaryResult, ['summary', 'xrefLinks'] ) ) );
+
+    return sources.length < 4 ? createInteractionsAppInfo(sources) : createEnrichmentAppInfo(sources);
+  }
+
+  render(){
+    let { geneResults } = this.props;
+    let { label, title, linkPath, description, imageClass } = this.determineAppLinkout();
+    let sources = geneResults.map( summaryResult => getHgncFromXref( _.get( summaryResult, ['summary', 'xrefLinks'] ) ) );
```
Comment (jvwong) [inline src/client/features/search/gene-results-view.js]: Just get `geneSymbol` for each.

File: src/client/features/search/gene-results-view.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      
+      if( sources.length === 1 ){
+        label = `View iteractions between ${sources.slice(0, sources.length - 1).join(', ')} and ${sources.slice(-1)}`;
+      }
+
+      return { label, title, linkPath, description, imageClass };
+    };
+
+    let { geneResults } = this.props;
+    let sources = geneResults.map( summaryResult => getHgncFromXref( _.get( summaryResult, ['summary', 'xrefLinks'] ) ) );
```
Comment (jvwong) [inline src/client/features/search/gene-results-view.js]: Just get `geneSymbol` for each.

**Comment by IgorRodchenkov** [comment]
@d2fong, scrolling is not on tour pictures but visible when I actually run the app

**Comment by IgorRodchenkov** [comment]
... the decision regarding when to show or hide the interaction tile is probably not good.

File: src/client/features/search/gene-results-view.js (hunk 2 above)
Comment (d2fong) [reply src/client/features/search/gene-results-view.js]: I dont see geneSymbol in the props object.  Can you be more specific?

Comment (d2fong) [reply src/client/features/search/gene-results-view.js]: These are the only things I see in the props object.  I don't see geneSymbol anywhere.
screen shot 2018-12-10 at 11 44 24 am

**Comment by d2fong** [comment]
@IgorRodchenkov I don't understand what you mean? Scrolling is not on my pictures but visible in the app?

File: src/client/features/search/gene-results-view.js (hunk 2 above)
Comment (d2fong) [reply src/client/features/search/gene-results-view.js]: Nevermind, I found it.

**Comment by d2fong** [comment]
> Just to reiterate our discussion there are two requirements: We need to convey what the app will actually provide (e.g. pathway enrichment) and the input we're are seeding that app with (i.e. HGNC gene IDs).
> 
> For multi-gene case, I was thinking about a setup where the genes are a thin strip, scrollable horizontally, and then maybe we finally use the margin for the 'app' panel, where we can put more of a full text description:
> 
> * Interactions
>   
>   * Blurb: `Interactions displays a network visualization of genes that interact with your query genes.`
>   * Button: `View genes that interact with X`
> * Enrichment
>   
>   * Blurb:`Enrichment displays a network visualization of Gene Ontology (Biological Process) and Reactome pathways that match some of the genes in your query.`
>   * Button: `View enriched pathways.`
> 
> Of course this is inspired by the Googles:
> 
> image

What do you mean by using the margin for the 'app panel'?

**Comment by d2fong** [comment]
also `Interactions displays a network visualization of genes that interact with your query genes` doesn't make sense for the multi gene case for interactions, because its only going to show the 2-3 source genes and their interactions.

**Comment by jvwong** [comment]
Yeah ok maybe let’s list the cases out for our templates

File: src/client/features/search/gene-results-view.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    let imageClass = 'interactions-logo';
+    let title = 'Interactions';
+
+    return { label, title, linkPath, description, imageClass };
+  }
+
+  render(){
+    let { geneResults } = this.props;
+    let sources = geneResults.map( geneInfo => geneInfo.geneSymbol );
+    let interactionsAppInfo = this.getInteractionsAppInfo( sources );
```
Comment (jvwong) [inline src/client/features/search/gene-results-view.js]: The `sources` arguments aren't used /declared in functions.

**Comment by d2fong** [comment]
Also included - Enrichment title changes:
screen shot 2018-12-13 at 10 48 53 am
screen shot 2018-12-13 at 10 48 47 am

**Comment by jvwong** [comment]
Also card app title could be more explicit. Consider 

- Interactions --> 'Gene interactions'
- Enrichment --> 'Pathway enrichment analysis'

**Pull request merged**